### ISSUE TITLE
`gauge_alg_test` fails with `--gridsize 3 2 1 1`

### ISSUE THREAD

**Issue by hummingtree** [lattice/quda#1097]
The following fails with `develop` branch:
```
mpirun -np 6 ./gauge_alg_test --prec single --dim 2 8 6 8 --gridsize 3 2 1 1
```
and the error message is:
```
[ RUN      ] GaugeAlgTest.Landau_Overrelaxation
Using curandStateMRG32k3a
Step 1
HB: Time = 0.045985 s, Gflop/s =    4.9, GB/s =    2.5
OVR: Time = 0.044191 s, Gflop/s =    5.2, GB/s =    2.4
Step 2
HB: Time = 0.045176 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043544 s, Gflop/s =    5.3, GB/s =    2.4
Step 3
HB: Time = 0.045442 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043509 s, Gflop/s =    5.3, GB/s =    2.4
Step 4
HB: Time = 0.044995 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043447 s, Gflop/s =    5.3, GB/s =    2.4
Step 5
HB: Time = 0.045225 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043448 s, Gflop/s =    5.3, GB/s =    2.4
Step 6
HB: Time = 0.045104 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043652 s, Gflop/s =    5.3, GB/s =    2.4
Step 7
HB: Time = 0.045124 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043437 s, Gflop/s =    5.3, GB/s =    2.4
Step 8
HB: Time = 0.045128 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043371 s, Gflop/s =    5.3, GB/s =    2.4
Step 9
HB: Time = 0.045235 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043782 s, Gflop/s =    5.2, GB/s =    2.4
Step 10
HB: Time = 0.045041 s, Gflop/s =    5.0, GB/s =    2.5
OVR: Time = 0.043429 s, Gflop/s =    5.3, GB/s =    2.4
Time Monte -> 0.906437 s
Plaq: 6.0884550087589395e-01, 6.0863874147751762e-01, 6.0905226027427017e-01
Landau gauge fixing with overrelaxation
Starting Landau gauge fixing...
	Overrelaxation boost parameter: 1.500000
	Stop criterium: 0.000000
	Stop criterium method: theta
	Maximum number of iterations: 100
	Reunitarize at every 10 steps
	Print convergence results at every 10 steps
Tuned block=(32,1,1), grid=(1,1,1), shared_bytes=0, aux=(1,1,1,1) giving 0.00 Gflop/s, 0.00 GB/s for cudaMemset with gaugefixingOVR,gauge_fix_ovr.cu,1092
Tuned block=(32,1,1), grid=(1,1,1), shared_bytes=0, aux=(1,1,1,1) giving 0.00 Gflop/s, 0.54 GB/s for cudaMemset with PreCalculateLatticeIndices,gauge_fix_ovr_extra.cu,84
Tuned block=(32,2,1), grid=(20,1,1), shared_bytes=3781, aux=(1,1,1,1) giving 53.74 Gflop/s, 148.11 GB/s for N4quda15GaugeFixQualityIfNS_5gauge11FloatNOrderIfLi18ELi2ELi18EL20QudaStaggeredPhase_s0ELb1EL19QudaGhostExchange_sn2147483648ELb0EEELi4EEE with vol=3456,stride=1728,precision=4,geometry=4,Nc=3
Step: 0	Action: -2.1697945101348320e-03	theta: 1.4274305138695738e+01
Tuned block=(32,1,1), grid=(1,1,1), shared_bytes=0, aux=(1,1,1,1) giving 0.00 Gflop/s, 0.00 GB/s for cudaMemcpyDeviceToHost with gaugefixingOVR,gauge_fix_ovr.cu,1188
Tuned block=(32,1,1), grid=(1,1,1), shared_bytes=0, aux=(1,1,1,1) giving 0.00 Gflop/s, 0.00 GB/s for cudaMemset with gaugefixingOVR,gauge_fix_ovr.cu,1194
Tuned , atomicadd=31), grid=(12,1,1), shared_bytes=2048, aux=(3,1,1,1) giving 85.09 Gflop/s, 40.54 GB/s for N4quda20GaugeFixBorderPointsIfNS_5gauge11FloatNOrderIfLi18ELi2ELi18EL20QudaStaggeredPhase_s0ELb1EL19QudaGhostExchange_sn2147483648ELb0EEELi4EEE with vol=3456,stride=1728,precision=4,geometry=4,Nc=3
ERROR: Auto-tuning failed for N4quda22GaugeFixInteriorPointsIfNS_5gauge11FloatNOrderIfLi18ELi2ELi18EL20QudaStaggeredPhase_s0ELb1EL19QudaGhostExchange_sn2147483648ELb0EEELi4EEE with vol=3456,stride=1728,precision=4,geometry=4,Nc=3 at vol=6x12x6x8 (rank 0, host yangyang, .../lib/targets/cuda/tune.cpp:826 in tuneLaunch())
       last kernel called was (name=N4quda22GaugeFixInteriorPointsIfNS_5gauge11FloatNOrderIfLi18ELi2ELi18EL20QudaStaggeredPhase_s0ELb1EL19QudaGhostExchange_sn2147483648ELb0EEELi4EEE,volume=6x12x6x8,aux=vol=3456,stride=1728,precision=4,geometry=4,Nc=3)

```

**Comment by weinbe2** [comment]
It looks like it's an issue with very small local volumes.

```
mpirun -np 6 ./gauge_alg_test --prec single --dim 6 8 6 8 --gridsize 3 2 1 1
```
Runs fine, while
```
mpirun -np 6 ./gauge_alg_test --prec single --dim 2 8 6 8 --gridsize 3 2 1 1
```
fails, with the issue noted above.

**Comment by weinbe2** [comment]
Oh, I see what's up:

```
ERROR: Auto-tuning failed for N4quda22GaugeFixInteriorPointsIfNS_5gauge11FloatNOrderIfLi18ELi2ELi18EL20QudaStaggeredPhase_s0ELb1EL19QudaGhostExchange_sn2147483648ELb0EEELi4EEE with vol=3456,stride=1728,precision=4,geometry=4,Nc=3 at vol=6x12x6x8 (rank 0, host prm-dgx-04, /gpfs/fs1/eweinberg/quda/2021-01-08GaugeAlgTest/quda/lib/targets/cuda/tune.cpp:826 in tuneLaunch())
```

It's tuning the **global** volume, not the local one: see `vol=6x12x6x8`.

**Comment by hummingtree** [comment]
@maddyscientist has the fix in #1056.

**Comment by weinbe2** [comment]
Closed in `develop` by #1074
